--- cfitsio-4.2.0/configure	2022-10-31 14:40:23.000000000 -0400
+++ cfitsio-4.2.0/configure	2023-07-14 11:46:53.298055665 -0400
@@ -744,6 +744,7 @@
 enable_reentrant
 enable_sse2
 enable_ssse3
+enable_standard_strings
 enable_symbols
 enable_hera
 with_bzip2
@@ -1386,6 +1387,8 @@
   --enable-sse2           Enable use of instructions in the SSE2 extended
                           instruction set
   --enable-ssse3          Enable use of instructions in the SSSE3 extended
+  --enable-standard-strings
+                          Enable use of FITSIO standard string processing
                           instruction set
   --enable-symbols        Enable debugging symbols by turning optimization off
   --enable-hera           Build for HERA (ASD use only)
@@ -2622,6 +2625,11 @@
 
 fi
 
+# Check whether --enable-standard_strings was given.
+if test "${enable_standard_strings+set}" = set; then :
+  enableval=$enable_standard_strings;  if test $enableval = yes; then USE_STANDARD_STRINGS=yes; fi
+
+fi
 
 SYMBOLS=""
 # Check whether --enable-symbols was given.
@@ -2646,6 +2654,14 @@
 
 fi
 
+# ------------------------------------------------------------------------------
+# Define FITS_USE_STANDARD_STRINGS
+# ------------------------------------------------------------------------------
+if test "x$USE_STANDARD_STRINGS" = xyes; then
+  $as_echo "#define FITS_USE_STANDARD_STRINGS 1" >>confdefs.h
+
+fi
+
 # Optional support for bzip2 compression:
 
 
@@ -15493,19 +15493,24 @@ fi
 printf "%s\n" "$ac_cv_lib_bz2_main" >&6; }
 if test "x$ac_cv_lib_bz2_main" = xyes
 then :
-  printf "%s\n" "#define HAVE_LIBBZ2 1" >>confdefs.h
+  LIBS="$LIBS -lbz2"; printf "%s\n" "#define HAVE_LIBBZ2 1" >>confdefs.h
+         for ac_header in bzlib.h
+do :
+  ac_fn_c_check_header_compile "$LINENO" "bzlib.h" "ac_cv_header_bzlib_h" "$ac_includes_default"
+if test "x$ac_cv_header_bzlib_h" = xyes
+then :
+  printf "%s\n" "#define HAVE_BZLIB_H 1" >>confdefs.h
 
-  LIBS="-lbz2 $LIBS"
+printf "%s\n" "#define HAVE_BZIP2 1" >>confdefs.h
 
-else case e in #(
-  e) as_fn_error $? "Unable to locate bz2 library needed when enabling bzip2 support; try specifying the path" "$LINENO" 5 ;;
-esac
+fi
+
+done
 fi
 
         else
             BZIP2_PATH="${withval}"
-        fi
-               for ac_header in bzlib.h
+                   for ac_header in bzlib.h
 do :
   ac_fn_c_check_header_compile "$LINENO" "bzlib.h" "ac_cv_header_bzlib_h" "$ac_includes_default"
 if test "x$ac_cv_header_bzlib_h" = xyes
@@ -15517,6 +15522,7 @@ printf "%s\n" "#define HAVE_BZIP2 1" >>confdefs.h
 fi
 
 done
+        fi
     fi
 
 
