--- cfitsio-4.2.0/configure.in	2022-10-31 14:40:23.000000000 -0400
+++ cfitsio-4.2.0/configure.in	2023-07-14 11:45:00.797390794 -0400
@@ -53,6 +53,12 @@
   [ if test $enableval = yes; then SSE_FLAGS="$SSE_FLAGS -mssse3"; fi ]
 )
 
+AC_ARG_ENABLE(
+  standard_strings,
+  [AS_HELP_STRING([--enable-standard-strings],[Enable use of FITSIO standard string processing])],
+  [ if test $enableval = yes; then USE_STANDARD_STRINGS=yes; fi ]
+)
+
 SYMBOLS=""
 AC_ARG_ENABLE(
   symbols,
@@ -71,6 +77,13 @@
   AC_DEFINE(BUILD_HERA)
 fi
 
+# ------------------------------------------------------------------------------
+# Define FITS_USE_STANDARD_STRINGS
+# ------------------------------------------------------------------------------
+if test "x$USE_STANDARD_STRINGS" = xyes; then
+  AC_DEFINE(FITS_USE_STANDARD_STRINGS)
+fi
+
 # Optional support for bzip2 compression:
 AC_ARG_WITH(
   bzip2,
