--- cfitsio-4.2.0/configure	2022-10-31 14:40:23.000000000 -0400
+++ cfitsio-4.2.0/configure	2023-07-14 11:46:53.298055665 -0400
@@ -744,6 +744,7 @@
 enable_reentrant
 enable_sse2
 enable_ssse3
+enable_standard_strings
 enable_symbols
 enable_hera
 with_bzip2
@@ -1386,6 +1387,8 @@
   --enable-sse2           Enable use of instructions in the SSE2 extended
                           instruction set
   --enable-ssse3          Enable use of instructions in the SSSE3 extended
+  --enable-standard-strings
+                          Enable use of FITSIO standard string processing
                           instruction set
   --enable-symbols        Enable debugging symbols by turning optimization off
   --enable-hera           Build for HERA (ASD use only)
@@ -2622,6 +2625,11 @@
 
 fi
 
+# Check whether --enable-standard_strings was given.
+if test "${enable_standard_strings+set}" = set; then :
+  enableval=$enable_standard_strings;  if test $enableval = yes; then USE_STANDARD_STRINGS=yes; fi
+
+fi
 
 SYMBOLS=""
 # Check whether --enable-symbols was given.
@@ -2646,6 +2654,14 @@
 
 fi
 
+# ------------------------------------------------------------------------------
+# Define FITS_USE_STANDARD_STRINGS
+# ------------------------------------------------------------------------------
+if test "x$USE_STANDARD_STRINGS" = xyes; then
+  $as_echo "#define FITS_USE_STANDARD_STRINGS 1" >>confdefs.h
+
+fi
+
 # Optional support for bzip2 compression:
 
 
