--- fitscore.c	2026-05-11 13:06:50.012650202 -0400
+++ fitscore.c	2026-05-14 13:13:16.392696912 -0400
@@ -4648,8 +4648,9 @@
 
         /* free the tile-compressed image cache, if it exists */
         if ((fptr->Fptr)->tilerow) {
-           ntilebins = 
-	    (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1;
+           // ntilebins = 
+	       //    (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1;
+           ntilebins = NTILEBINS;
 
            for (ii = 0; ii < ntilebins; ii++) {
              if ((fptr->Fptr)->tiledata[ii]) {
@@ -4702,8 +4703,9 @@
         /* free the tile-compressed image cache, if it exists */
         if ((fptr->Fptr)->tilerow) {
 
-           ntilebins = 
-	    (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1;
+           /* ntilebins =  */
+           /*      (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1; */
+           ntilebins = NTILEBINS;
 
            for (ii = 0; ii < ntilebins; ii++) {
              if ((fptr->Fptr)->tiledata[ii]) {
@@ -4814,8 +4816,9 @@
      /* free the tile-compressed image cache, if it exists */
      if ((fptr->Fptr)->tilerow) {
 
-           ntilebins = 
-	    (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1;
+           /* ntilebins =  */
+           /*     (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1; */
+           ntilebins = NTILEBINS;
 
            for (ii = 0; ii < ntilebins; ii++) {
              if ((fptr->Fptr)->tiledata[ii]) {
@@ -5023,8 +5026,9 @@
      /* free the tile-compressed image cache, if it exists */
      if ((fptr->Fptr)->tilerow) {
 
-           ntilebins = 
-	    (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1;
+           /* ntilebins =  */
+           /*     (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1; */
+           ntilebins = NTILEBINS;
 
            for (ii = 0; ii < ntilebins; ii++) {
              if ((fptr->Fptr)->tiledata[ii]) {
@@ -6713,8 +6717,9 @@
           /* free the tile-compressed image cache, if it exists */
           if ((fptr->Fptr)->tilerow) {
 
-           ntilebins = 
-	    (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1;
+           /* ntilebins =  */
+           /*     (((fptr->Fptr)->znaxis[0] - 1) / ((fptr->Fptr)->tilesize[0])) + 1; */
+           ntilebins = NTILEBINS;
 
            for (ii = 0; ii < ntilebins; ii++) {
              if ((fptr->Fptr)->tiledata[ii]) {
