--- /home/esheldon/Downloads/cfitsio-4.2.0/fitsio.h	2022-10-31 14:40:23.000000000 -0400
+++ /home/esheldon/Downloads/cfitsio-4.2.0-mod/fitsio.h	2023-07-14 11:48:10.102506229 -0400
@@ -811,6 +811,7 @@
 /*---------------- utility routines -------------*/
 
 float CFITS_API ffvers(float *version);
+int CFITS_API fits_use_standard_strings(void);
 void CFITS_API ffupch(char *string);
 void CFITS_API ffgerr(int status, char *errtext);
 void CFITS_API ffpmsg(const char *err_message);
