--- cfitsio3430/fitsio.h	2018-03-01 10:28:51.000000000 -0600
+++ cfitsio3430/fitsio.h	2018-12-14 08:39:20.000000000 -0600
@@ -797,6 +797,7 @@
 /*---------------- utility routines -------------*/

 float CFITS_API ffvers(float *version);
+int CFITS_API fits_use_standard_strings(void);
 void CFITS_API ffupch(char *string);
 void CFITS_API ffgerr(int status, char *errtext);
 void CFITS_API ffpmsg(const char *err_message);
