--- cfitsio-4.2.0/fitscore.c	2022-10-31 14:40:23.000000000 -0400
+++ cfitsio-4.2.0/fitscore.c	2023-07-14 11:47:54.870417064 -0400
@@ -194,6 +194,21 @@
 
     return(*version);
 }
+
+
+/*
+   Return 1 if we are to treat strings per the FITS standard (not
+   replacing nulls with spaces, and not padding with spaces)
+*/
+int fits_use_standard_strings(void) {
+#ifdef FITS_USE_STANDARD_STRINGS
+    return 1;
+#else
+    return 0;
+#endif
+}
+
+
 /*--------------------------------------------------------------------------*/
 int ffflnm(fitsfile *fptr,    /* I - FITS file pointer  */
            char *filename,    /* O - name of the file   */
