--- cfitsio-4.2.0/fitsio.h	2022-10-31 14:40:23.000000000 -0400
+++ cfitsio-4.2.0/fitsio.h	2023-07-14 11:48:10.102506229 -0400
@@ -811,6 +810,10 @@
 /*---------------- utility routines -------------*/

 float CFITS_API ffvers(float *version);
+#ifndef _FITSIO_H_FITS_USE_STANDARD_STRINGS
+#define _FITSIO_H_FITS_USE_STANDARD_STRINGS
+int CFITS_API fits_use_standard_strings(void);
+#endif
 void CFITS_API ffupch(char *string);
 void CFITS_API ffgerr(int status, char *errtext);
 void CFITS_API ffpmsg(const char *err_message);
