--- imcompress.c	2026-05-11 13:06:45.276636041 -0400
+++ imcompress.c	2026-05-15 09:29:32.853352044 -0400
@@ -1818,7 +1818,9 @@
     if ((outfptr->Fptr)->tilerow) {  /* has the tile cache been allocated? */
 
       /* calculate the column bin of the compressed tile */
-      tilecol = (row - 1) % ((long)(((outfptr->Fptr)->znaxis[0] - 1) / ((outfptr->Fptr)->tilesize[0])) + 1);
+      // tilecol = (row - 1) % ((long)(((outfptr->Fptr)->znaxis[0] - 1) / ((outfptr->Fptr)->tilesize[0])) + 1);
+      // Cache only a single tile
+      tilecol = (row - 1) % NTILEBINS;
       
       if ((outfptr->Fptr)->tilerow[tilecol] == row) {
         if (((outfptr->Fptr)->tiledata)[tilecol]) {
@@ -5949,7 +5951,8 @@
     if ((infptr->Fptr)->tilerow == 0)  {
 
       /* calculate number of column bins of compressed tile */
-      ntilebins =  (((infptr->Fptr)->znaxis[0] - 1) / ((infptr->Fptr)->tilesize[0])) + 1;
+      // ntilebins =  (((infptr->Fptr)->znaxis[0] - 1) / ((infptr->Fptr)->tilesize[0])) + 1;
+      ntilebins = NTILEBINS;
 
      if ((infptr->Fptr)->znaxis[0]   != (infptr->Fptr)->tilesize[0] ||
         (infptr->Fptr)->tilesize[1] != 1 ) {   /* don't cache the tile if only single row of the image */
@@ -5967,7 +5970,9 @@
     /* check if this tile was cached; if so, just copy it out */
     if ((infptr->Fptr)->tilerow)  {
       /* calculate the column bin of the compressed tile */
-      tilecol = (nrow - 1) % ((long)(((infptr->Fptr)->znaxis[0] - 1) / ((infptr->Fptr)->tilesize[0])) + 1);
+      // tilecol = (nrow - 1) % ((long)(((infptr->Fptr)->znaxis[0] - 1) / ((infptr->Fptr)->tilesize[0])) + 1);
+      // Cache only a single tile
+      tilecol = (nrow - 1) % NTILEBINS;
 
       if (nrow == (infptr->Fptr)->tilerow[tilecol] && datatype == (infptr->Fptr)->tiletype[tilecol] ) {
 
