--- cfitsio-4.2.0/putcols.c	2022-10-31 14:40:24.000000000 -0400
+++ cfitsio-4.2.0/putcols.c	2023-07-14 11:49:03.954820768 -0400
@@ -157,7 +157,16 @@
 
          for (;jj < twidth; jj++)    /* fill field with blanks, if needed */
          {
-           *buffer = ' ';
+           if ( fits_use_standard_strings() ) {
+               if (snull[0] == ASCII_NULL_UNDEFINED) {
+                   *buffer = ' ';
+               } else {
+                   *buffer = '\0';
+               }
+           } else {
+               *buffer = ' ';
+           }
+
            buffer++;
          }
 
